File: src/Romarr.Core/MediaFiles/RomFile.cs
using System;
using System.Collections.Generic;
using Romarr.Common.Extensions;
using Romarr.Core.Datastore;
using Romarr.Core.Languages;
using Romarr.Core.MediaFiles.MediaInfo;
using Romarr.Core.Parser.Model;
using Romarr.Core.Qualities;
using Romarr.Core.Games;

namespace Romarr.Core.MediaFiles
{
    public class RomFile : ModelBase
    {
        public int GameId { get; set; }
        public int PlatformNumber { get; set; }
        public string RelativePath { get; set; }
        public string Path { get; set; }
        public long Size { get; set; }
        public DateTime DateAdded { get; set; }
        public string OriginalFilePath { get; set; }
        public string SceneName { get; set; }
        public string ReleaseGroup { get; set; }
        public string ReleaseHash { get; set; }
        public QualityModel Quality { get; set; }
        public IndexerFlags IndexerFlags { get; set; }
        public MediaInfoModel MediaInfo { get; set; }
        public LazyLoaded<List<Rom>> Roms { get; set; }
        public LazyLoaded<Game> Game { get; set; }
        public List<Language> Languages { get; set; }
        public string Region { get; set; }
        public string CrcHash { get; set; }
        public ReleaseType ReleaseType { get; set; }
        public RomFileType RomFileType { get; set; }
        public string PatchVersion { get; set; }
        public string DlcIndex { get; set; }
        public int? LinkedGameId { get; set; }
        public string Revision { get; set; }
        public int DumpQuality { get; set; }
        public int Modification { get; set; }
        public string ModificationName { get; set; }
        public int RomReleaseType { get; set; }

        public override string ToString()
        {
            return string.Format("[{0}] {1}", Id, RelativePath);
        }

        public string GetSceneOrFileName()
        {
            if (SceneName.IsNotNullOrWhiteSpace())
            {
                return SceneName;
            }

            if (RelativePath.IsNotNullOrWhiteSpace())
            {
                return System.IO.Path.GetFileNameWithoutExtension(RelativePath);
            }

            if (Path.IsNotNullOrWhiteSpace())
            {
                return System.IO.Path.GetFileNameWithoutExtension(Path);
            }

            return string.Empty;
        }
    }
}

File: frontend/src/RomFile/RomFile.ts
import ModelBase from 'App/ModelBase';
import ReleaseType from 'InteractiveImport/ReleaseType';
import Language from 'Language/Language';
import { QualityModel } from 'Quality/Quality';
import CustomFormat from 'typings/CustomFormat';
import MediaInfo from 'typings/MediaInfo';

export interface RomFile extends ModelBase {
  gameId: number;
  platformNumber: number;
  relativePath: string;
  path: string;
  size: number;
  dateAdded: string;
  sceneName: string;
  releaseGroup: string;
  languages: Language[];
  region?: string;
  crcHash?: string;
  quality: QualityModel;
  customFormats: CustomFormat[];
  customFormatScore: number;
  indexerFlags: number;
  releaseType: ReleaseType;
  romFileType: number;
  patchVersion?: string;
  dlcIndex?: string;
  revision?: string;
  dumpQuality: number;
  modification: number;
  modificationName?: string;
  romReleaseType: number;
  mediaInfo: MediaInfo;
  qualityCutoffNotMet: boolean;
}
